app/org/sagebionetworks/bridge/services/AuthenticationService.java

```diff
@@ -238,6 +254,40 @@ public void resetPassword(PasswordReset passwordReset) throws BridgeServiceExcep
         accountDao.resetPassword(passwordReset);
     }
     
+    private void repairConsents(Study study, Account account, UserSession session, ClientInfo clientInfo){
+        boolean repaired = false;
+        
+        Map<SubpopulationGuid,ConsentStatus> statuses = session.getUser().getConsentStatuses();
+        for (Map.Entry<SubpopulationGuid,ConsentStatus> entry : statuses.entrySet()) {
+            ConsentSignature activeSignature = account.getActiveConsentSignature(entry.getKey());
```

Comment:
Verify if account.getActiveConsentSignature() works for Stormpath accounts that have the old consent_signature key but not the new consent_signatureS key.